### PR TITLE
support callable python object as statement

### PR DESCRIPTION
**Pull request by chaudum**
Example for a benchmark that adds 100 columns using a statement
generating callable:

```py
class ColumnGenerator:

    def __init__(self, prefix):
        self.prefix = prefix
        self.stmt = 'alter table t_add_column add column "{}_{}" string'
        self.count = 0

    def __call__(self):
        self.count += 1
        return self.stmt.format(self.prefix, self.count)

    def __str__(self):
        return self.stmt

spec = Spec(
    setup=Instructions(
        statements=[...]
    ),
    teardown=Instructions(
        statements=[...]
    ),
    queries=[
        {
            'statement': ColumnGenerator('str_col'),
            'iterations': 100,
        },
    ]
)
```

### CODE REVIEW

File: cr8/clients.py [hunk 1]
```diff
@@ -42,6 +42,10 @@ def _to_http_hosts(hosts: Union[Iterable[str], str]) -> List[str]:
         return r['duration']
 
 
+def plain_or_callable(obj):
+    return obj and callable(obj) and obj() or obj
```
Comment (mfussenegger) [inline cr8/clients.py]: I think `obj() if callable(obj) else obj` would be more readable. Or `obj if not callable(obj) else obj()` if you'd prefer that.
And please prefix the method with `_`

Comment (chaudum) [inline cr8/clients.py]: ok!

File: cr8/clients.py [hunk 2]
```diff
@@ -42,8 +42,31 @@ def _to_http_hosts(hosts: Union[Iterable[str], str]) -> List[str]:
...
+    >>> _plain_or_callable(args):
+    [1, 'name']
+
+    >>> class BulkArgsGenerator:
+    ...     def __call__(self):
+    ...         return [[1, 'foo'], [2, 'bar'], [3, 'foobar']]
+    >>> _plain_or_callable(BulkArgsGenerator())
+    [[1, 'foo'], [2, 'bar'], [3, 'foobar']]
+    """
+    return obj() if obj and callable(obj) else obj
```
Comment (mfussenegger) [inline cr8/clients.py]: `if obj` is not necessary, callable(None) is false anyway
```
>>> foo = None
>>> callable(foo)
False
>>> foo() if callable(foo) else foo
>>> 
```

**Comment by mfussenegger** [comment]
Merged https://github.com/mfussenegger/cr8/commit/5c3051ada4b876466eb8a5ffba0307a10c986e7d & https://github.com/mfussenegger/cr8/commit/1aeb7dd75a3aefde00614c95781b1ab1d6e5aaf7

**Pull request closed without merging**